### PR TITLE
[FLINK-35281][hotfix][cdc-common] FlinkEnvironmentUtils#addJar add each jar only once

### PR DESCRIPTION
**Pull request by loserwang1024**
Current org.apache.flink.cdc.composer.flink.FlinkEnvironmentUtils#addJar will be invoked for each source and sink.

```java
public static void addJar(StreamExecutionEnvironment env, URL jarUrl) {
    try {
        Class<StreamExecutionEnvironment> envClass = StreamExecutionEnvironment.class;
        Field field = envClass.getDeclaredField("configuration");
        field.setAccessible(true);
        Configuration configuration = ((Configuration) field.get(env));
        List<String> jars =
                configuration.getOptional(PipelineOptions.JARS).orElse(new ArrayList<>());
        jars.add(jarUrl.toString());
        configuration.set(PipelineOptions.JARS, jars);
    } catch (Exception e) {
        throw new RuntimeException("Failed to add JAR to Flink execution environment", e);
    } 

```
if multiple source or sink share same jar, the par path will be added repeatly. For example, mysql to mysql later.

### CODE REVIEW

**Comment by loserwang1024** [comment]
@yuxiqian , @PatrickRen , CC

File: flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/flink/FlinkEnvironmentUtils.java [hunk 1]
```diff
@@ -22,29 +22,48 @@
...
      */
     public static void addJar(StreamExecutionEnvironment env, URL jarUrl) {
+        addJar(env, Lists.newArrayList(jarUrl));
+    }
+
+    /**
+     * Add the specified JARs to {@link StreamExecutionEnvironment} so that the JAR will be uploaded
+     * together with the job graph.
+     */
+    public static void addJar(StreamExecutionEnvironment env, List<URL> jarUrls) {
```
Comment (yuxiqian) [inline flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/flink/FlinkEnvironmentUtils.java:54]:
```suggestion
    public static void addJar(StreamExecutionEnvironment env, Collection<URL> jarUrls) {
```

File: flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/flink/FlinkEnvironmentUtils.java [hunk 2]
```diff
@@ -22,29 +22,48 @@
...
             Configuration configuration = ((Configuration) field.get(env));
-            List<String> jars =
-                    configuration.getOptional(PipelineOptions.JARS).orElse(new ArrayList<>());
-            jars.add(jarUrl.toString());
-            configuration.set(PipelineOptions.JARS, jars);
+            Set<String> jars =
+                    configuration.getOptional(PipelineOptions.JARS).orElse(new ArrayList<>())
+                            .stream()
+                            .collect(Collectors.toSet());
+            jars.addAll(jarUrls.stream().map(URL::toString).collect(Collectors.toList()));
```
Comment (yuxiqian) [inline flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/flink/FlinkEnvironmentUtils.java:64]: Converting it to `Set` might lost orderliness of input. What about using `stream().distinct()` instead?

**Review by yuxiqian — COMMENTED** [review]
Thanks for @loserwang1024's contribution, I just left some minor comments.

File: flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/flink/FlinkEnvironmentUtils.java (hunk 1 above)
Comment (loserwang1024) [reply flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/flink/FlinkEnvironmentUtils.java]: done it

**Review by yuxiqian — APPROVED** [review]

File: flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/flink/FlinkEnvironmentUtils.java [hunk 3]
```diff
@@ -22,29 +22,49 @@
...
 import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 /** Utilities for {@link org.apache.flink.streaming.api.environment.StreamExecutionEnvironment}. */
 @Internal
 public class FlinkEnvironmentUtils {
```
Comment (ruanhang1993) [inline flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/flink/FlinkEnvironmentUtils.java:39]: Would you mind to add a private constructor for this util class?

**Review by ruanhang1993 — APPROVED** [review]

**Pull request merged**